src/commandHandlers/stats.ts

```diff
@@ -0,0 +1,20 @@
+import { Client, MessageEmbed } from 'discord.js';
+
+import { StatsService } from './stats.service';
+
+export const command = async (client: Client, embed: MessageEmbed) => {
+    const statsService = new StatsService(client)
```

Comment:
Perhaps this solution could be improved in the future, in order to only instantiate the service once. Surely it's not a problem now to create a new instance every time a user types the `stats` command, but could there be performance or memory issues later on?
What do you think @eddiejaoude ? I'm trying to be careful to not pre-optimize what doesn't need optimization 😅 .

Reply:
You are right, a new instance isn't needed each time. A singleton might be better in this case 🤔

Reply:
Yes, creating a singleton in this file (`stats.ts`) would be easy if we `export` the client object in `index.ts`, like @garretcharp suggested.
Or we could also create the singleton in `index.ts` and `export` that instead.

My only concern with this approach is having "global" variables (since these objects will be available to get imported **anywhere**) but in I don't see a problem doing it right now

Reply:
I would think exporting the client would be the easiest route to go, it allows us to create any new handlers which require the client super easily because we can just import it rather than bounce it around functions. 

The main thing if anything is the `embed` should be the first argument in the command as it is the one that every command relies on where as client is currently only used in one command.

Also we shouldn't make every command an `async` function only use it when needed. All of the commandHandler files should not need to change only the new command and the `commands.ts`. 

We should also be able to remove the extra await that is before the message send in the `command.ts` file.

So:
```js
message.channel.send(await matching.command(client, embed));
```

Rather than:
```js
await message.channel.send(await matching.command(client, embed));
```

Reply:
Great feedback. I am doing it in stages so I don't break it all 😂 .

I moved `client` to its own file and can be imported in the command that requires it rather than passing it around - great suggestion 👍

Reply:
I made the `StatsService` into a singleton